interview-questions.md

```diff
@@ -9,63 +9,121 @@ Then, research the question to expand on your answer. Even if you feel you have
 
   Your answer:
 
+  MVC stands for module view controller, basic pieces of a full stack app.
+
   Researched answer:
 
+  MVC stands for module (database), View, whate the user sees (HTML,CSS, React). A full stack application has all the three pieces that can come from different technologies, an example of a decoupled application is when M and C comes from Rails and teh V is developed in React.
 
 
 2. What does it mean if an application has CRUD? What does CRUD stand for?
 
   Your answer:
 
+  C.R.U.D Create Read Update and Delete. they are the 4 basic funnctions/interactions involved into internet.
+
   Researched answer:
 
+  C.R.U.D :
+  Create or add new data;
+  Read, retrieve, search, or view existing data;
+  Update or edit existing data;
+  Delete, deactivate, or remove existing data. 
 
 
 3. What is a relational database? Are there other kinds of databases?
 
   Your answer:
 
+  Relational dabtes are dabatases where everything is stored in rows and columns.
+
   Researched answer:
 
+  A relational database is a collection of data items with pre-defined relationships between them.
+   Items are organized as a set of tables with columns and rows. 
+  Tables are used to hold information about the objects to be represented in the database. 
+  Each column in a table holds a certain kind of data and a field stores the actual value. 
+  The rows in the table represent a collection of related values of one object. 
+  Each row in a table is marked with a unique identifier (id).
+  There are a 11 main types od databases available on the market, depending on the needs.
+
 
 
 4. What are the 5 HTTP verbs? What are they important?
 
   Your answer:
 
+  POST, GET, PUT, PATCH and DELETE. they are the main five berbs of CRUD operations to make a website/app usable
+
   Researched answer:
 
+  the 5 HTTP verbs are:
+
+  1 Get, to retrieve data,
+  2 Post, to submit data and trigger a change
+  3 Put, to replace current representations with the one loaded
+  4 Patch, to aply partial modifications to a resorce
+  5 Delete, to delete a specified resource 
 
 
 5. What is object-relational mapping?
 
   Your answer:
 
+  n/a
+
   Researched answer:
 
+  A programming tecnique to convert data between imcompatible sysems using OOP object oriented programming. 
+  ORM is time saving because data is written only in one place and it's easier to update and mantain, it's written in the coding language o your choice and it let;s you use OOP with its data inheritance.
 
 
 6. What is a gem?
 
   Your answer:
 
+  Gem is a packaging manager for Ruby.
```

Comment:
A Gem is a package of Ruby code. Whereas "RubyGems" is the package manager or the place where all open source gems live.